packages/salesforcedx-vscode-core/test/jest/services/sdr/componentSetUtils.test.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ConfigUtil } from '@salesforce/salesforcedx-utils-vscode';
+import { ComponentSet } from '@salesforce/source-deploy-retrieve';
+import * as vscode from 'vscode';
+import { WorkspaceContext } from '../../../../src/context/workspaceContext';
+import { setApiVersionOn } from '../../../../src/services/sdr/componentSetUtils';
+
+jest.mock('@salesforce/salesforcedx-utils-vscode', () => {
+  return {
+    ...jest.requireActual('@salesforce/salesforcedx-utils-vscode'),
+    TelemetryService: { getInstance: jest.fn() },
+    ChannelService: jest.fn().mockImplementation(() => {
+      return {};
+    })
+  };
+});
+
+jest.mock('../../../../src/messages', () => {
+  return { loadMessageBundle: jest.fn(), nls: { localize: jest.fn() } };
+});
+
+describe('componentSetUtils', () => {
+  const userConfigApiVersion = '49.0';
+  const orgApiVersion = '56.0';
+  let createFileSystemWatcherMock: jest.SpyInstance;
+  let getUserConfiguredApiVersionMock: jest.SpyInstance;
+  let workspaceContextGetConnectionMock: jest.SpyInstance;
+  const mockConnection = {
+    getApiVersion: jest.fn().mockReturnValue(orgApiVersion)
+  } as any;
+  const mockWatcher = {
+    onDidChange: jest.fn(),
+    onDidCreate: jest.fn(),
+    onDidDelete: jest.fn()
+  };
+
+  beforeEach(() => {
+    workspaceContextGetConnectionMock = jest
+      .spyOn(WorkspaceContext.prototype, 'getConnection')
+      .mockResolvedValue(mockConnection);
+    createFileSystemWatcherMock = (vscode.workspace
+      .createFileSystemWatcher as any).mockReturnValue(mockWatcher);
```

Comment:
interesting this mock was necessary

Reply:
@gbockus-sf it is not needed anymore now that this branch is parented by #4682.  I'm going to remove it before merging this FYI.